lib/lutaml/model/xml_adapter/xml_document.rb

```diff
@@ -70,12 +70,24 @@ def build_options_for_nested_elements(options = {})
           options
         end
 
-        def parse_element(element)
+        def parse_element(element, klass = nil, format = nil)
           result = Lutaml::Model::MappingHash.new
           result.item_order = element.order
 
           element.children.each_with_object(result) do |child, hash|
-            value = child.text? ? child.text : parse_element(child)
+            attr = klass.attribute_for_child(child.name, format) if klass&.<= Serialize
+
+            value = if child.text?
+                      child.text
+                    elsif attr&.raw?
+                      child.children.map do |c|
+                        next c.text if c.text?
+
+                        self.class.new(c).to_xml
```

Comment:
@HassanAkbar let's merge this first, but is it possible to obtain the full raw string instead of serializing it again? I just don't want to modify something that is meant to be "raw".

Reply:
@ronaldtse the conversion in `lutaml-model` has 2 steps
1. Data is parsed using the adapter and saved in the adapter document class (at this point we have no idea which classes have which mappings defined)
2. Mappings are used to convert that data to ruby object

The problem is that when we are parsing the data we do not have access to the mappings so we have no idea which content to read as raw string and which to parse. So that is why we are converting the parsed data back to XML string when applying mappings.

Reply:
Can’t the parser return us the raw XMLstring at a certain node?

Reply:
@ronaldtse It will require some changes in the adapters but I think it will work something like 

```ruby
value = if child.text?
          child.text
        elsif attr&.raw?
          child.to_xml
        else
          parse_element(child, attr&.type || klass, format)
        end
```

I'll open a separate PR for this.